### PR TITLE
Weldertank ruptured sprite

### PR DESCRIPTION
**Pull request by NightmareChamillian**
## About the PR 

Adds a sprite so it's more obvious when weldertanks are ruptured. Also sorry for bad master branch (I have since fixed it but not this branch)

image

## Why's this needed? 

Too many times, I have been a lonely engineer in need of fuel and have walked up to what I thought was a perfectly functional fueltank to discover some trickster had exploded it.

## Changelog
```changelog
(u)NightmarechaMillian
(+) Fueltanks, when they blow up, will now have a sprite change. Blown up fueltanks also count as open containers.
```

### CODE REVIEW

File: code/modules/chemistry/tools/dispensers.dm [hunk 1]
```diff
@@ -360,6 +360,15 @@
 		else if (ismob(usr))
 			add_fingerprint(usr, TRUE)
 
+	smash()
+		..()
+		icon_state = "weldtank1"
```
Comment (pali6) [inline code/modules/chemistry/tools/dispensers.dm:365]: `smash` deletes the object anyway, is this necessary?

File: code/modules/chemistry/tools/dispensers.dm [hunk 2]
```diff
@@ -360,6 +360,15 @@
...
+
+	ex_act(severity)
+		..()
+		icon_state = "weldtank1" //to ensure that a weldertank's always going to be updated by their own explosion
```
Comment (pali6) [inline code/modules/chemistry/tools/dispensers.dm:369]: I think naming the icon state something like `weldtank-broken` or `weldtank-exploded` would be more readable than `weldtank1`.

**Comment by BatElite** [comment]
Is the tank still capable of holding reagents after it explodes?

(I'm not aware of a way to transfer to them but I wanna say there was something)

**Comment by NightmareChamillian** [comment]
> Is the tank still capable of holding reagents after it explodes?

Beyond the sprites, nothing's been enabled or disabled about the exploded fueltank, which means it retains all functionality

**Review by WALPVRGIS — APPROVED** [review]
LGTM on the art side of things. 👍

**Comment by frawhst** [comment]
Can exploding it make it an open container. fits the sprite

**Review by ZeWaka — CHANGES_REQUESTED** [review]
x

**Comment by NightmareChamillian** [comment]
Some things:
a) had my plate quite full this past week, apologies for leaving this to dry for so long
b) They're now open containers, but have some weird interactions due to requiring a click-drag in order to pour reagents into them

File: code/modules/chemistry/tools/dispensers.dm [hunk 3]
```diff
@@ -360,6 +361,16 @@
 		else if (ismob(usr))
 			add_fingerprint(usr, TRUE)
 
+	smash()
+		..()
+
```
Comment (ZeWaka) [inline code/modules/chemistry/tools/dispensers.dm:366]:
```suggestion
```

**Review by ZeWaka — COMMENTED** [review]
i don't see any open container code

File: code/modules/chemistry/tools/dispensers.dm [hunk 4]
```diff
@@ -360,6 +361,16 @@
...
+	ex_act(severity)
+		..()
+		icon_state = "weldtank-burst" //to ensure that a weldertank's always going to be updated by their own explosion
+		isburst = TRUE
+
+	is_open_container()
+		return isburst
```
[the remark this answers is not in the document]
Comment (NightmareChamillian) [reply code/modules/chemistry/tools/dispensers.dm:373]: I don't know much about open container code, but provided my understanding is correct, this should return 1 if it's burst, which lets whatever called the proc know that it's an open container

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**